python/cudf/cudf/core/dataframe.py

```diff
@@ -1339,8 +1339,14 @@ def _slice(self: T, arg: slice) -> T:
 
     @_cudf_nvtx_annotate
     def memory_usage(self, index=True, deep=False):
-        return Series(
-            {str(k): v for k, v in super().memory_usage(index, deep).items()}
+        mem_usage = [col.memory_usage for col in self._data.columns]
+        names = [str(name) for name in self._data.names]
+        if index:
+            mem_usage.append(self._index.memory_usage())
+            names.append("Index")
+        return Series._from_data(
+            data={None: as_column(mem_usage)},
+            index=as_index(names),
         )
```

Comment:
Does Dask actually need these to live on the device? Given that this is typically small, maybe we should just keep the values on host?

Reply:
We have optimized the dask dispatch itself `sizeof_cudf_dataframe`, which is 200x faster now. So this code will not get called.

Reply:
@shwina this is what @galipremsagar and I discussed offline. We're limited in how far we can optimize `memory_usage` since it needs to return a `Series` to match pandas. We could keep it on the host, but even creating a pandas Series is not cheap (relatively speaking):
```
In [2]: %timeit pd.Series({'a': 1})
114 µs ± 2.23 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```
so for the purpose of the Dask registration doing the "manual" calculation that we've implemented now is a much bigger and more worthwhile win. If you'd like we could also change `memory_usage` to return a pandas Series instead of a cudf one, which would be a little odd but has some precedent in what we do for `df.columns`, but I don't know if it's worth it unless the ~2x speedup in that API is meaningful for some specific non-Dask workflow.

Reply:
Very nice. Glad you found a better place to optimize!